sqlitedict.py

```diff
@@ -390,19 +395,34 @@ def __init__(self, filename, autocommit, journal_mode):
         self.reqs = Queue()
         self.setDaemon(True)  # python2.5-compatible
         self.exception = None
+        self.initialized = None
+        self.timeout = timeout
         self.log = logging.getLogger('sqlitedict.SqliteMultithread')
         self.start()
 
     def run(self):
-        if self.autocommit:
-            conn = sqlite3.connect(self.filename, isolation_level=None, check_same_thread=False)
-        else:
-            conn = sqlite3.connect(self.filename, check_same_thread=False)
-        conn.execute('PRAGMA journal_mode = %s' % self.journal_mode)
-        conn.text_factory = str
-        cursor = conn.cursor()
-        conn.commit()
-        cursor.execute('PRAGMA synchronous=OFF')
+        try:
+            if self.autocommit:
+                conn = sqlite3.connect(self.filename, isolation_level=None, check_same_thread=False)
+            else:
+                conn = sqlite3.connect(self.filename, check_same_thread=False)
+        except Exception as ex:
+            self.log.exception("Failed to initialize connection for filename: %s" % self.filename)
+            self.exception = (e_type, e_value, e_tb) = sys.exc_info()
```

Comment:
What's the point of declaring the new variables?

```suggestion
            self.exception = sys.exc_info()
```